### PR TITLE
614: Refactor repositories API calls

### PR DESCRIPTION
**Pull request by katiefenn**
- Replaced fetch repositories API calls with call-api middleware

### CODE REVIEW

File: src/common/actions/repositories.js [hunk 1]
```diff
@@ -1,66 +1,29 @@
...
 export function fetchUserRepositories(pageNumber) {
-  return async (dispatch) => {
-    let urlParts = [BASE_URL, 'api/github/repos'];
+  let urlParts = ['/api/github/repos'];
 
-    dispatch({
-      type: REPOSITORIES_REQUEST
-    });
+  if (pageNumber) {
+    urlParts.push('page/' + pageNumber);
```
Comment (bartaz) [inline src/common/actions/repositories.js]: It's not really part of this PR, but I wonder why we send params to this one API via url parts where in every other API we use query.

File: test/routes/src/server/routes/t_github.js [hunk 2]
```diff
@@ -640,12 +641,21 @@ const hasStatus = (expected) => {
   };
 };
 
-const hasMessage = (expected) => {
+const hasPayloadMessage = (expected) => {
```
Comment (bartaz) [inline test/routes/src/server/routes/t_github.js]: Why these are called `hasPayloadMessage` `hasBodyMessage` when they seem to be checking `code` property?

File: src/common/reducers/repositories.js [hunk 3]
```diff
@@ -20,14 +20,14 @@ export function repositories(state = {
...
-        pageLinks: action.payload.pageLinks
+        ids: union(state.ids, action.payload.response.result),
+        pageLinks: action.payload.response.pageLinks
       };
     case ActionTypes.REPOSITORIES_FAILURE:
       return {
         ...state,
         isFetching: false,
-        error: action.payload,
+        error: action.payload.error,
```
Comment (bartaz) [inline src/common/reducers/repositories.js]: With `callApi` middleware we need to nest error in payload instead of error just being the payload?

I guess it's needed so we can make sure we also pass any other payload items if needed, right?

It breaks a little bit Flux Standard Actions pattern, but I guess it's a compromise we need to make. I think we were already breaking FSA in this regard to send both `id` and `error` in payload. So I guess it needs to stay like that.

So I guess in our FSA 'variant' we should assume that if `action.error` is true then `action.payload.error` should be an error object.

File: src/common/actions/repositories.js (hunk 1 above)
Comment (bartaz) [inline src/common/actions/repositories.js]: Or maybe - since there are small updated to serverside of this API in this PR, maybe it can be updated to query be consistent with rest of API calls?

Comment (katiefenn) [inline src/common/actions/repositories.js]: Yeah, I think it's worth a bit of time.

File: src/common/reducers/repositories.js (hunk 3 above)
Comment (katiefenn) [inline src/common/reducers/repositories.js]: Yes, this is the compromise that I made.

FSA _requires_ that `action.error` is a bool, but passing the error as the payload itself is only a _convention_.

File: test/routes/src/server/routes/t_github.js (hunk 2 above)
Comment (katiefenn) [inline test/routes/src/server/routes/t_github.js]: `hasPayloadMessage` is checking for a code property inside `action.payload` whereas `hasBodyMessage` is checking for a code property at the top level of `action`. This is part of the gradual removal of the `payload` property from API responses, and when it has been removed across the board `hasPayloadMessage` can be removed.

I think it's worth doing this work in stages, but open to discussion about whether a different approach should be taken.
